src/ape_test/provider.py

```diff
@@ -133,8 +134,15 @@ def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
         receipt = self.get_receipt(
             txn_hash.hex(), required_confirmations=txn.required_confirmations or 0
         )
-        if txn.gas_limit is not None and receipt.ran_out_of_gas:
-            raise OutOfGasError()
+
+        if receipt.failed:
+            # Replay txn to get revert reason
+            try:
+                txn_dict = receipt.transaction.dict()
+                txn_dict["nonce"] += 1
+                self.web3.eth.call(cast(TxParams, txn_dict))
+            except (ValidationError, TransactionFailed) as err:
+                raise self.get_virtual_machine_error(err, sender=txn.sender) from err
```

Comment:
Do the `ValidationError` and `TransactionFailed` only get raised via `web3.eth.call()`? If so I think we can just put that line only in the `try` so it doesn't imply that the other two lines could raise either of those